File: src/main/java/es/us/dp1/l4_01_24_25/upstream/matchTile/MatchTile.java
package es.us.dp1.l4_01_24_25.upstream.matchTile;

import es.us.dp1.l4_01_24_25.upstream.coordinate.Coordinate;
import es.us.dp1.l4_01_24_25.upstream.match.Match;
import es.us.dp1.l4_01_24_25.upstream.model.BaseEntity;
import es.us.dp1.l4_01_24_25.upstream.tile.Tile;
import es.us.dp1.l4_01_24_25.upstream.tile.TileType;
import es.us.dp1.l4_01_24_25.upstream.validation.ValidNumber;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name="match_tile")
public class MatchTile extends BaseEntity {

    @ValidNumber(min=0, max=5)
    Integer capacity;

    @ValidNumber(min=0)
    Integer orientation;

    @ValidNumber(min=0, max=5)
    Integer salmonsNumber;

    @Embedded
    Coordinate coordinate;

    @ManyToOne
    Tile tile;

    @ManyToOne
    Match match;

    public Boolean isFull () {
        if (this.match.getPlayersNumber() >= 2 && this.getTile().getType().equals(TileType.ROCK)) return this.getCapacity().equals(this.getSalmonsNumber() -1);
        else return this.getCapacity().equals(this.getSalmonsNumber()); 
    }

}

File: frontend/src/interfaces/MatchTile.ts
import Coordinate from "./Coordinate.ts";
import Tile from './Tile.ts';

export default interface MatchTile {
  id: number;
  /** * @min 0 * @max 5 */
  capacity: number;
  /** * @min 0 */ 
  orientation: number;
  /** * @min 0 * @max 5 */
  salmonsNumber: number;
  coordinate: Coordinate;
  tile: Tile;
  matchId: number;
}

export const emptyTile: MatchTile = { id: 0, capacity: 0, orientation: 0, salmonsNumber: 0, coordinate: { x: 0, y: 0 }, tile: { id: 0, image:'', type: '' }, matchId: 0 };